### PR TITLE
Add clipboard handling

### PR DESCRIPTION
**Pull request by blink1073**
Allow for programmatic access to copy and paste events.

### CODE REVIEW

File: src/clipboard.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+  // Execute the command.
+  document.execCommand(type);
+
+  // Restore the previous selection.
+  sel = window.getSelection();
+  sel.removeAllRanges();
+  for (var i = 0, len = savedRanges.length; i < len; ++i) {
+    sel.addRange(savedRanges[i]);
+  }
```
Comment (sccolbert) [inline src/clipboard.ts:63]: I don't think this is what you want. Why are you changing the selection before triggering the command?

Comment (blink1073) [inline src/clipboard.ts:63]: The intent is to select the node of interest for the copy event, then restore the previous selection.

Comment (blink1073) [inline src/clipboard.ts:63]: So the desired node receives the event.

Comment (sccolbert) [inline src/clipboard.ts:63]: Does it need to be selected, or does it just need be focused?

Comment (blink1073) [inline src/clipboard.ts:63]: It has to be selected.

Comment (sccolbert) [inline src/clipboard.ts:63]: This feels wrong to me. But whatever. We can fix it later if it becomes a problem.

**Pull request merged**